src/Tribe/Rewrite.php

```diff
@@ -632,32 +632,48 @@ public function filter_rewrite_parse_query_vars( array $query_vars = [], array $
 	 *               an entry..
 	 */
 	protected function get_option_controlled_slug_entry( array $localized_matchers, $default_slug, $option_name ) {
-		$using_default_archive_slug = $default_slug === tribe_get_option( $option_name, $default_slug );
+		$current_slug       = tribe_get_option( $option_name, $default_slug );
+		$using_default_slug = $default_slug === $current_slug;
 
 		$filter = static function ( $matcher ) use ( $default_slug )
 		{
 			return isset( $matcher['query_var'] )
 			       && 'post_type' === $matcher['query_var']
 			       && isset( $matcher['localized_slugs'] )
-			       && is_array( $matcher['localized_slugs'] )
-			       && in_array( $default_slug, $matcher['localized_slugs'], true );
+			       && is_array( $matcher['localized_slugs'] );
 		};
 
-		$archive_localized_matcher = array_filter( $localized_matchers, $filter );
-		$archive_localized_matcher = reset( $archive_localized_matcher );
+		$target_matcher = array_filter( $localized_matchers, $filter );
+		$target_matcher = reset( $target_matcher );
 
-		if ( $using_default_archive_slug || false === $archive_localized_matcher ) {
+		if ( $using_default_slug || false === $target_matcher ) {
 			return [];
 		}
 
-		$archive_localized_matcher['localized_slugs'][] = $archive_localized_matcher['en_slug'];
-		// Create an entry for each localized slug to replace (?:events).
+		/**
+		 * Add the slugs in the following order: default slug, option-controlled slug, localized slug.
+		 */
+		array_unshift( $target_matcher['localized_slugs'], $default_slug, $current_slug );
+
+		// Make sure we do not have duplicated slugs.
+		$target_matcher['localized_slugs'] = array_unique( $target_matcher['localized_slugs'] );
+
+		// Create a replacement string that contains all of them.
+		$all_slugs = array_unique( array_reverse( $target_matcher['localized_slugs'] ) );
+
 		$entry = [
-			'(?:' . $default_slug . ')' => [
+			// Create an entry for the localized slug to replace `(?:events)`.
+			'(?:' . $default_slug . ')'              => [
+				'query_var'       => 'post_type',
+				'en_slug'         => $target_matcher['en_slug'],
+				'localized_slugs' => $target_matcher['localized_slugs']
+			],
+			// Create an entry for the localized slug to replace `(?:events|foo|bar)`.
+			'(?:' . implode( '|', $all_slugs ) . ')' => [
 				'query_var'       => 'post_type',
-				'en_slug'         => $archive_localized_matcher['en_slug'],
-				'localized_slugs' => $archive_localized_matcher['localized_slugs']
-			]
+				'en_slug'         => $target_matcher['en_slug'],
+				'localized_slugs' => $target_matcher['localized_slugs']
```

Comment:
* :no_entry_sign: Each line in an array declaration must end in a comma
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: 259a7dedbde1a59988c1b12ebc6944ad</sub>